src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/SampleServiceImpl.java

```diff
@@ -363,10 +378,29 @@ public Map<String, List<Long>> getSampleIdsBySampleNameForProjects(List<Long> pr
 	public void removeSequencingObjectFromSample(Sample sample, SequencingObject object) {
 		SampleSequencingObjectJoin readObjectForSample = ssoRepository.readObjectForSample(sample, object.getId());
 		ssoRepository.delete(readObjectForSample);
-		if (sample.getDefaultSequencingObject() != null
-				&& sample.getDefaultSequencingObject().getId().equals(object.getId())) {
+		if (sample.getDefaultSequencingObject() != null && sample.getDefaultSequencingObject()
+				.getId()
+				.equals(object.getId())) {
 			sampleRepository.removeDefaultSequencingObject(sample);
 		}
+		Set<AnalysisSubmission> submissions = submissionRepository.findAnalysisSubmissionsForSequencingObject(object);
+		if (submissions.isEmpty() && object.getSequencingRun() == null) {
```

Comment:
**nitpick (blocking):** this if statement could use a bit more comments and clarification of existing comments as it appears to me to do two separate actions. First action to remove the to be deleted sequencingObject as a source from the ConcatedSequencingObject table and then second to remove the entry from the ConcatedSequencingObject table if the sequencingObject is the result of a concation. I think this would be the perfect opportunity to split these 2 actions into separate private methods that are called within the if statement, These methods could be called `removeSequencingObjectAsConcationSource` and `removeConcatedSequencingObjectForSequencingObject`, these methods would do the checking to see that they exist before removal.

Reply:
Is this closer to what you're thinking? Please see 3ab13c9a9ff66ba5167e98fb01f51a8980388548.

Reply:
That is really close, can you just remove the extra s in the second method which I am guessing was a typo. I.e. `removeConcatedSequencingObjectsForSequencingObject` should be `removeConcatedSequencingObjectForSequencingObject`. Otherwise it looks great and should be alot more readable.

Reply:
Please see c75342412341b1d8ecdc600668060d6cb20b7179.